src/containers/PatientProfilePage.tsx

```diff
@@ -92,6 +99,11 @@ const PatientProfilePage = ({
       });
     }
   }, [data]);
+  useEffect(() => {
+    if (mode === "new" && barcodeValue) {
+      setFormFields({ ...formFields, barcodeValue });
+    }
+  }, []);
```

Comment:
if the dependency array is empty (e.g. `[]`), this will only run once (whenever this component does get rendered) - is this what you intended?

Reply:
yeah it's intended